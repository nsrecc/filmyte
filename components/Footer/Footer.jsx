import React from 'react';
import styles from './Footer.module.scss';

const Footer = () => {
  return (
    <footer className={styles.footer}>
      <p id="copyright">&copy; 2021 filmyte.com</p>
      <div className={styles.poweredBy}>
        <p className={styles.poweredByText}>Powered by:</p>
        <ul className={styles.poweredByList}>
          <li>
            <a href="https://vercel.com/" target="_blank" rel="noopener noreferrer">
              <img src="/vercel.svg" alt="Vercel Logo" className={styles.logo} />
            </a>
          </li>
          <li>
            <a href="https://www.themoviedb.org/" target="_blank" rel="noopener noreferrer">
              The Movie Database (TMDb)
            </a>
          </li>
        </ul>
      </div>
    </footer>
  );
};

export default Footer;
