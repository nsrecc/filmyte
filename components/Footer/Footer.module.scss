@use "styles/variables" as *;

.footer {
  width: 100%;
  padding: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  // background-color: $ui-color-background-secondary;
  background-image: linear-gradient($ui-color-background-primary, $ui-color-background-secondary);

  // powered by section
  .poweredBy {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 1rem;

    .poweredByText {
      padding-left: 2rem;
    }

    .poweredByList {
      display: inline-flex;
      list-style: none;
      padding: 0;
      margin: 0 auto;
  
      // powered by list items
      li {
        padding: 0.5rem 2rem;
        border-right: 1px solid $gray-50;

        a {
          padding: 0.5rem;
          display: flex;
          justify-content: center;
          align-items: center;
        }
  
        &:last-of-type {
          border-right: 0;
        }
      }
    }
  }

  > *:last-child {
    border-top: 1px solid $gray-50;
  }
}

// TO DO: rename / change
.logo {
  height: 1.3em;
}
