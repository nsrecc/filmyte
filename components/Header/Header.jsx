/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import Link from 'next/link'; // Wraps <a> elements
import { useThemeContext } from 'providers/ThemeProvider/ThemeProvider';
import { Sun, Moon } from 'phosphor-react';
import styles from './Header.module.scss';

const Header = () => {
  const { themeIsLight, toggleTheme } = useThemeContext();

  // Props for dark mode button icon
  const iconProps = {
    color: themeIsLight ? '#000000' : '#ffffff',
    size: '1.5em',
    weight: 'bold',
  };

  /* eslint-disable react/jsx-props-no-spreading */
  const themeIcon = themeIsLight ? <Moon {...iconProps} /> : <Sun {...iconProps} />;
  /* eslint-enable react/jsx-props-no-spreading */

  return (
    <header className={styles.header}>
      <div className={styles.headerInner}>
        <Link href="/">
          <a className={styles.headerLogoLink}>
            <span className={styles.filmyteLogo}>Filmyte</span>
          </a>
        </Link>
        <div className={styles.headerEnd}>
          <nav className={styles.navWrapper}>
            <ul className={styles.navList}>
              <li>
                <Link href="/search">
                  <a>Search</a>
                </Link>
              </li>
              <li>
                <Link href="/discover">
                  <a>Discover</a>
                </Link>
              </li>
              <li>
                <Link href="/about">
                  <a>About</a>
                </Link>
              </li>
            </ul>
          </nav>
          <button
            className={styles.themeButton}
            type="button"
            id="theme-toggle"
            onClick={toggleTheme}
          >
            {themeIcon}
          </button>
        </div>
      </div>
    </header>
  );
};

export default Header;
