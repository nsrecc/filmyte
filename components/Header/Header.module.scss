@use "styles/variables" as *;
@use "styles/mixins" as m;

.header {
  width: 100%;
  padding: 1rem 2rem;
  display: inline-flex;
  align-items: center;
  background-color: $filmyte-rajah;

  .headerLogoWrapper {
    margin-right: 1.5rem;
    padding: 0.5rem;
    display: flex;
    justify-content: center;
    align-items: center;
    
    .filmyteLogo {
      @include m.filmyteLogo;
      font-size: 2em;
    }
  }

  .navWrapper {
    .navList {
      display: inline-flex;
      list-style: none;
      padding: 0;
      margin: 0 auto;

      // nav list items
      li {
        margin: 0 1.5rem;

        a {
          padding: 0.5rem; // added padding to increase mouse click area
          font-size: 1.5em;
          font-weight: bold;
        }
      }
    }
  }
}
