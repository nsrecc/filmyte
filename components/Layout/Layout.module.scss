@use "styles/mixins" as m;

.main {
  @include m.maxWidthWrapper;
}
