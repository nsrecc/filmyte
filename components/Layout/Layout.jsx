import React from 'react';
import Head from 'next/head';
import Header from 'components/Header/Header';
import Footer from 'components/Footer/Footer';
// import styles from './Layout.scss';

const Layout = ({ children }) => {
  return (
    <>
      <Head>
        {/* <meta charset="utf-8" /> */}
        <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
        <meta name="viewport" content="width=device-width, initial-scale=1" />

        <title>Filmyte | Discover Your Movies and TV Shows</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Header />
      <main>
        { children }
      </main>
      <Footer />
    </>
  );
};

export default Layout;
