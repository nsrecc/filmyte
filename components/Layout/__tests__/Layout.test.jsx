/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Header from 'components/Header/Header';
import Footer from 'components/Footer/Footer';
import Layout from '../Layout';

describe('Layout', () => {
  let props;

  beforeEach(() => {
    props = {
      children: <p>test</p>,
    };
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('contains default document title', () => {
    const defaultTitle = 'Filmyte | Discover Your Movies and TV Shows';

    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper.find('title').exists()).toBe(true);
    expect(wrapper.find('title').text()).toMatch(defaultTitle);
  });

  it('contains Header', () => {
    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper.find(Header).exists()).toBe(true);
  });

  it('contains main content with children', () => {
    props.children = <h1>test child</h1>;

    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper.contains(
      <main>
        { props.children }
      </main>
    ));
  });

  it('contains Footer', () => {
    const wrapper = shallow(<Layout {...props} />);
    expect(wrapper.find(Footer).exists()).toBe(true);
  });
});
