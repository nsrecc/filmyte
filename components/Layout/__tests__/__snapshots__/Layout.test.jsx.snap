// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<Fragment>
  <Head>
    <meta
      content="IE=edge"
      httpEquiv="X-UA-Compatible"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <title>
      Filmyte | Discover Your Movies and TV Shows
    </title>
    <link
      href="/favicon.ico"
      rel="icon"
    />
    <link
      href="https://fonts.gstatic.com"
      rel="preconnect"
    />
    <link
      href="https://fonts.googleapis.com/css2?family=Asap:ital,wght@0,400;1,700&family=Open+Sans&display=swa"
      rel="stylesheet"
    />
  </Head>
  <Header />
  <main
    className="mainContainer"
  >
    <p>
      test
    </p>
  </main>
  <Footer />
</Fragment>
`;
